### DISCUSSION TITLE
AttributeError: module 'graphlearn' has no attribute 'encoders'

### DISCUSSION THREAD

**Discussion by zdx** [alibaba/GraphScope#122] [Q&A] [answered]
**Describe the bug**
tensorflow :2.2.0

run node_classification_on_citation.ipynb :
config = {"class_num": 349, # output dimension
            "features_num": 130, # 128 dimension + kcore + triangle count
            "batch_size": 500,
            "val_batch_size": 100,
            "test_batch_size":100,
            "categorical_attrs_desc": "",
            "hidden_dim": 256,
            "in_drop_rate": 0.5,
            "hops_num": 2,
            "neighs_num": [5, 10],
            "full_graph_mode": False,
            "agg_type": "gcn",  # mean, sum
            "learning_algo": "adam",
            "learning_rate": 0.0005,
            "weight_decay": 0.000005,
            "epoch": 20,
            "node_type": "paper",
            "edge_type": "cites"}

train(config, lg)

the error:
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-33-d7e8d35bad41> in <module>
     18             "edge_type": "cites"}
     19 
---> 20 train(config, lg)

<ipython-input-32-900f9818c2bd> in train(config, graph)
     27                              config["learning_algo"],
     28                              config["learning_rate"],
---> 29                              config["weight_decay"]))
     30     trainer.train_and_evaluate()

~/anaconda3/envs/graphscope/lib/python3.7/site-packages/graphscope/learning/graphlearn/python/model/tf/trainer.py in __init__(self, model_func, epoch, optimizer)
    161                                          epoch,
    162                                          optimizer)
--> 163     self.model = self._model_func()
    164 
    165   def init(self, **kwargs):

<ipython-input-32-900f9818c2bd> in model_fn()
     21                     node_type=config["node_type"],
     22                     edge_type=config["edge_type"],
---> 23                     full_graph_mode=config["full_graph_mode"])
     24     trainer = LocalTFTrainer(model_fn,
     25                              epoch=config["epoch"],

~/anaconda3/envs/graphscope/lib/python3.7/site-packages/graphscope/learning/examples/tf/gcn/gcn.py in __init__(self, graph, output_dim, features_num, batch_size, val_batch_size, test_batch_size, categorical_attrs_desc, hidden_dim, in_drop_rate, hops_num, neighs_num, full_graph_mode, node_type, edge_type)
     88     self.src_ego_spec = gl.EgoSpec(src_spec, hops_spec=[hop_spec] * self.hops_num)
     89     # encoders.
---> 90     self.encoders = self._encoders()
     91 
     92   def _sample_seed(self):

~/anaconda3/envs/graphscope/lib/python3.7/site-packages/graphscope/learning/examples/tf/gcn/gcn.py in _encoders(self)
    128 
    129     depth = self.hops_num
--> 130     feature_encoders = [gl.encoders.IdentityEncoder()] * (depth + 1)
    131     conv_layers = []
    132     for i in range(depth):

AttributeError: module 'graphlearn' has no attribute 'encoders'

**Environment (please complete the following information):**
 - GraphScope version: main
 - OS: Linux
 - Version centos7
 - Kubernetes Version:1.20.1

**Additional context**

**Comment by sighingnow** [comment, marked as answer]
Looks like you encounter a tensorflow's `ImportError` first, then install tensorflow and continue to run the demo script.

To use models from `examples/` in `graphscope.learning.examples`, you need to install **tensorflow** first then "import graphscope", i.e., you need to restart the notebook after install tensorflow.

**Comment by sighingnow** [comment]
Hi @zdx  I think that has already been fixed in your environment. Feel free to re-open if you have any further comment.
